src/js/plugins/plugin.js

```diff
@@ -37,6 +37,9 @@ function getJSPath(url) {
 
 const Plugin = function(url) {
     this.url = url;
+    this.promise = new Promise((resolve) => {
```

Comment:
Can we go without this - seems a bit of an anti-pattern.

Reply:
I don't like the circular reference, but this was the most elegant way to ensure all plugins have a promise that resolves when `this.js` is available.